examples/data_preprocessing/police_killing/preprocessing_rules_from_paper.ipynb

```diff
@@ -0,0 +1,1009 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Starting with simple rules based on the keywords of the paper.\n",
+    "\n",
+    "\"These lists were semi-automatically constructed by looking up the nearest neighbors of “police” and “kill” (by cosine distance) from Google’s public release of word2vec vectors pretrained on a very large (proprietary) Google News corpus,20 and then manually excluding a small number of misspelled words or redundant capitalizations (e.g. “Police” and “police”).\" (Keith et al. p. 11)\n",
+    "\n",
+    "I saved the keywords in a csv."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 36,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def get_keywords(data_path):\n",
+    "    keywords = pd.read_csv(os.path.join(data_path, \"keywords.csv\"))\n",
+    "    return(keywords)\n",
+    "\n",
+    "keywords = get_keywords(data_path)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 37,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#dictionary mapping the rules to their ID\n",
+    "\n",
+    "def rule2id(keywords):\n",
+    "    \n",
+    "    rule2rule_id = dict({})\n",
+    "    rule_id = 0\n",
+    "    for police_word in keywords[\"police_words\"]:\n",
+    "        for kill_word in keywords[\"kill_words\"].dropna():\n",
+    "            rule2rule_id[f'{police_word} {kill_word}'] = rule_id\n",
+    "            rule_id += 1\n",
+    "    \n",
+    "    return(rule2rule_id)\n",
+    "\n",
+    "rule2rule_id = rule2id(keywords)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 38,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#getting the rule ID and creating a new dict with it, assigning \n",
+    "#the value 1 to all of them because there is only one class\n",
+    "\n",
+    "def rule_id2label_id(rule2rule_id):\n",
+    "\n",
+    "    rule2label = dict({})\n",
+    "\n",
+    "    for rule, rule_id in rule2rule_id.items():\n",
+    "        rule2label[rule_id] = 1\n",
+    "        \n",
+    "    return(rule2label)\n",
+    "        \n",
+    "\n",
+    "rule2label = rule_id2label_id(rule2rule_id)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 39,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\"\"\"\n",
+    "manually creating a label2label_id dict, since there are only two classes, \n",
+    "and a label_id2label dict (I will need that later, but I'm not sure if I'm allowed to to it like this)\n",
+    "Is it okay to create it manually?\n",
+    "\"\"\"\n",
+    "\n",
+    "label2label_id ={\"negative\":0, \"positive\":1}\n",
+    "label_id2label = {0: \"negative\", 1: \"positive\"}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## building the T matrix\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 40,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "num_classes = 2"
```

Comment:
should be somewhere in the beginning of the notebook where you give info about the dataset